sw/device/tests/csrng_edn_concurrency_test.c

```diff
@@ -394,8 +402,21 @@ static void entropy_config(void) {
   // Wait for the esfinal FIFO inside ENTROPY_SRC to fill up. This will reduce
   // the wait time for ENTROPY_SRC seeds and instead increase contention on the
   // CSRNG and EDN hardware pipelines which are in focus for this test.
+  //
+  // Under normal operating conditions, producing one seed takes roughly 4 ms.
+  // We wait for at most 32 ms for 6 seeds (2 for the EDNs, 1 for CSRNG, 3 to
+  // fill the esfinal FIFO). In the reduced_freq varaint of the test, raw
+  // entropy may be generated at a lower rate. Depending on the configuration of
+  // the EDNs, the constant entropy consumption of AST may prevent the esfinal
+  // FIFO from filling up.
   dif_entropy_src_debug_state_t debug_state;
+  uint32_t iter_count = 0;
   do {
+    busy_spin_micros(kEsFinalFifoIterationTimeUs);
```

Comment:
The code looks ok but I think using `ibex_timeout_{init,check}` would be a bit cleaner.

Reply:
Thanks @pamaury . In the interest of time, I leave this as is but I will investigate the `ibex_timeout_{init,check}` option next time I need something similar. Thanks for the pointer.